src/web/js/cpo-main.js

```diff
@@ -267,19 +299,30 @@
             }, 0);
             return ret.promise;
           },
-          run: function(str, name) {
+          run: function(str, name, lineNo) {
             var ret = Q.defer();
-            setTimeout(function() {
+            setTimeout(function () {
               runtime.runThunk(function() {
-                return runtime.safeCall(
-                  function() {
-                    return gf(repl,
-                    "make-interaction-locator").app(
-                      runtime.makeFunction(function() { return str; }))
-                  },
-                  function(locator) {
+                //console.log('doing run timeout');
+                if (cpoDialect!=='patch') {
```

Comment:
Sure. If using a switch statement, would you prefer Pyret to simply be the default: case, or would you like a specific label ('pyret') for it?  (My preference is that Pyret simply be the default, since it's presumably always going to be the "default" dialect, in that it won't require any translation layer. It's a purely aesthetic choice to be sure.)

Reply:
I might probably write it as 
```js
switch(cpoDialect) { 
case "Patch": 
  ...patch specific stuff...;
  break;
case "Pyret":
default:
  ...Pyret specific stuff
}
```
and get the best of both worlds :)